packages/features/handleMarkNoShow.ts

```diff
@@ -139,17 +118,143 @@ const handleMarkNoShow = async ({ bookingUid, attendees, noShowHost }: TNoShowIn
         },
       });
 
-      responsePayload["noShowHost"] = true;
-      responsePayload["message"] = "No-show status updated";
+      responsePayload.setNoShowHost(true);
+      responsePayload.setMessage("No-show status updated");
     }
 
-    return responsePayload;
+    return responsePayload.getPayload();
   } catch (error) {
     if (error instanceof Error) {
       logger.error(error.message);
     }
-    return { message: "Failed to update no-show status" };
+    throw new HttpError({ statusCode: 500, message: "Failed to update no-show status" });
   }
 };
 
+const getAttendees = async (
+  bookingUid: string,
+  attendeeEmails: string[],
+  attendees: NonNullable<TNoShowInputSchema["attendees"]>
+) => {
+  const allAttendees = await prisma.attendee.findMany({
+    where: {
+      AND: [
+        {
+          booking: {
+            uid: bookingUid,
+          },
+          email: {
+            in: attendeeEmails,
+          },
+        },
+      ],
+    },
+    select: {
+      id: true,
+      email: true,
+    },
+  });
+
+  const allAttendeesMap = allAttendees.reduce((acc, attendee) => {
+    acc[attendee.email] = attendee;
+    return acc;
+  }, {} as Record<string, { id: number; email: string }>);
+
+  const updatePromises = attendees.map((attendee) => {
+    const attendeeToUpdate = allAttendeesMap[attendee.email];
+    if (!attendeeToUpdate) return;
+    return prisma.attendee.update({
+      where: { id: attendeeToUpdate.id },
+      data: { noShow: attendee.noShow },
+    });
+  });
+
+  const results = await Promise.allSettled(updatePromises);
+  logFailedResults(results);
+
+  return results
+    .filter((x) => x.status === "fulfilled")
+    .map((x) => (x as PromiseFulfilledResult<{ noShow: boolean; email: string }>).value)
+    .map((x) => ({ email: x.email, noShow: x.noShow }));
+};
+
+const getWebhooksService = async (bookingUid: string) => {
+  const booking = await prisma.booking.findUnique({
+    where: { uid: bookingUid },
+    select: {
+      id: true,
+      eventType: {
+        select: {
+          id: true,
+          teamId: true,
+          userId: true,
+        },
+      },
+    },
+  });
+
+  const orgId = await getOrgIdFromMemberOrTeamId({
+    memberId: booking?.eventType?.userId,
+    teamId: booking?.eventType?.teamId,
+  });
+  const webhooks = await new WebhookService({
+    teamId: booking?.eventType?.teamId,
+    userId: booking?.eventType?.userId,
+    eventTypeId: booking?.eventType?.id,
+    orgId,
+    triggerEvent: WebhookTriggerEvents.BOOKING_NO_SHOW_UPDATED,
+  });
+
+  return { webhooks, bookingId: booking?.id };
+};
+
+const checkCanAccessBooking = async (bookingUid: string, userId?: number) => {
```

Comment:
I feel we should have `assert` in function name as it throws on failure which is what an assert do.
```suggestion
const assertCanAccessBooking = async (bookingUid: string, userId?: number) => {
```